### DISCUSSION TITLE
I want to work on this project for my thesis, how can I contribute?

### DISCUSSION THREAD

**Discussion by idoan** [TinyMPC/TinyMPC#15] [Ideas]
Hi guys, I already hold a BSc in Control and Computer science. After a long gap, This year I returned to the university and this semester I will write a thesis. I have been searching for a suitable topic and this seems to be a good candidate, which encompasses both software and control topics.

Do you think this library could be a good topic for thesis? Would you recommend it for this purpose? And most importantly, do you think this library is complete or will it include other core features? like adding optimal control etc? or on the contrary, do you think only required developments are subtle ones like creating new the demos for tutorials in each codebase?

And I am planning on getting crazyflie 2.1 to work on this codebase. Do I need any additional sensors/peripheral/units/attachments/deck/antennas etc. to use this code?

Thanks

**Comment by xkhainguyen** [comment]
Hi there,

Thanks for reaching out! Making MPC work well on embedded systems is definitely an interesting topic. 

TinyMPC is made for real-time optimal control. We are also in the process of improving this framework as well as adding more useful features. Stay tuned for more updates from us! Also, we recommend deploying TinyMPC for different robots and different tasks.

Regarding the Crazyflie, you can start using TinyMPC with minimal hardware including the base quadrotor 2.1, Crazyradio, and Flow deck v2.

We welcome all pull requests or contributions from the community.

Good luck!

**Reply by idoan** [reply]
Thanks for the reply. Is there a roadmap or something like a roadmap covering the main features you would like to add in the near future?

**Reply by xkhainguyen** [reply]
You can take a look at future work in our paper [https://arxiv.org/abs/2310.16985](https://arxiv.org/abs/2310.16985). It should be representative of what we are working on. We hope to release the next major update by March.
